src/modules/lists/lists.controller.ts

```diff
@@ -70,4 +70,41 @@ export class ListsController {
     };
   }
 
+  @ApiOperation({ title: 'Updates a given list' })
+  @Put('/:listid')
+  @UsePipes(new ValidationPipe({ transform: true, whitelist: true }))
+  async updateList(@Req() request: Request, @Param('userid') userId: string, @Param('listid') listId: string, @Body() data: ListDto) {
+    const current: User = await this.usersService.findByAuth0Id(request.user.auth0Id);
+    const user: User = await this.usersService.findById(userId);
+
+    // check if user exists
+    if (!user) {
+      throw new BadRequestException('User not found');
+    }
+
+    // only admins and current user can update lists
+    if (!current._id.equals(user._id) && !current.roles.includes(Role.ADMIN)) {
+      throw new UnauthorizedException('Not authorized to perform this operation');
+    }
+
+    const list: List[] = await this.listsService.findByUserId({userId, listId});
```

Comment:
I think you are forgetting setting the `isFavorite: false` filter here, we don't want to update the favorite list ;)